File: packages/host-wire/src/limits.ts
export const PROTOCOL_VERSION = "omp-app/1" as const;
/** Schema compatibility version, intentionally independent from the T4 release package version. */
export const APP_WIRE_VERSION = "0.7.0" as const;
export const MAX_INPUT_BYTES = 1_048_576;
export const MAX_TRANSCRIPT_LINE_BYTES = 10_485_760;
export const MAX_STRING_BYTES = 65_536;
export const MAX_ID_BYTES = 256;
export const MAX_ARRAY_ITEMS = 1_000;
export const MAX_MAP_KEYS = 512;
export const MAX_JSON_DEPTH = 32;
export const MAX_JSON_NODES = 20_000;
export const MAX_CAPABILITIES = 128;
export const MAX_TERMINAL_OUTPUT_BYTES = 256_000;
export const MAX_FILE_BYTES = 768 * 1024;
/** Raw bytes carried by one base64 image-upload command. */
export const IMAGE_UPLOAD_CHUNK_BYTES = 256 * 1024;
export const IMAGE_UPLOAD_CHUNK_BASE64_BYTES = Math.ceil(IMAGE_UPLOAD_CHUNK_BYTES / 3) * 4;
export const IMAGE_UPLOAD_MAX_BYTES = 20 * 1024 * 1024;
export const PROMPT_IMAGE_MAX_COUNT = 8;
/** Raw bytes returned by one transcript-image read command. */
export const TRANSCRIPT_IMAGE_CHUNK_BYTES = 256 * 1024;
export const TRANSCRIPT_IMAGE_CHUNK_BASE64_BYTES = Math.ceil(TRANSCRIPT_IMAGE_CHUNK_BYTES / 3) * 4;
export const TRANSCRIPT_IMAGE_MAX_BYTES = IMAGE_UPLOAD_MAX_BYTES;
export const TRANSCRIPT_IMAGE_MAX_COUNT = 64;
/** Maximum descriptors retained on one durable entry. */
export const MAX_ARTIFACTS_PER_ENTRY = 64;
/** Maximum file summaries retained for one turn review. */
export const MAX_TURN_FILE_CHANGES = 4_096;
/** Raw bytes returned by one generic artifact read command. */
export const ARTIFACT_CHUNK_BYTES = 256 * 1024;
export const ARTIFACT_CHUNK_BASE64_BYTES = Math.ceil(ARTIFACT_CHUNK_BYTES / 3) * 4;
export const ARTIFACT_MAX_BYTES = 20 * 1024 * 1024;
/** Raw bytes returned by one preview-capture read command. */
export const PREVIEW_CAPTURE_CHUNK_BYTES = 256 * 1024;
export const PREVIEW_CAPTURE_CHUNK_BASE64_BYTES = Math.ceil(PREVIEW_CAPTURE_CHUNK_BYTES / 3) * 4;
export const PREVIEW_CAPTURE_MAX_BYTES = 8 * 1024 * 1024;
export const PREVIEW_CAPTURE_MAX_PIXELS = 16 * 1024 * 1024;
export const PREVIEW_MAX_PER_SESSION = 8;
export const PREVIEW_TEXT_INPUT_BYTES = 8 * 1024;
export const PREVIEW_SCROLL_DELTA_LIMIT = 100_000;
export const PREVIEW_SELECTOR_BYTES = 4 * 1024;
export const PREVIEW_HANDOFF_MESSAGE_BYTES = 2 * 1024;
export const PREVIEW_LEASE_TTL_MAX_MS = 10 * 60 * 1000;
export const PREVIEW_HANDOFF_TIMEOUT_MAX_MS = 10 * 60 * 1000;
export const MAX_SAVED_CURSORS = 128;
export const MAX_EPOCH_BYTES = 128;

File: apps/ios/HostWire/Sources/HostWire/Limits.swift
import Foundation

/// Protocol-wide constants (host-wire/src/limits.ts). Kept as a namespace.
public enum Wire {
    /// The single protocol id carried by every frame's `v` field.
    public static let protocolVersion = "omp-app/1"
    /// Schema compatibility version, independent of the T4 package version.
    public static let appWireVersion = "0.7.0"
}

/// Numeric limits from host-wire/src/limits.ts. Names mirror the TS constants.
public enum Limits {
    public static let maxInputBytes = 1_048_576
    public static let maxStringBytes = 65_536
    public static let maxIdBytes = 256
    public static let maxArrayItems = 1_000
    public static let maxMapKeys = 512
    public static let maxCapabilities = 128
    public static let maxSavedCursors = 128
    public static let maxEpochBytes = 128

    public static let imageUploadChunkBytes = 256 * 1024
    public static let imageUploadMaxBytes = 20 * 1024 * 1024
    public static let promptImageMaxCount = 8

    public static let transcriptImageChunkBytes = 256 * 1024
    public static let transcriptImageMaxCount = 64

    public static let maxArtifactsPerEntry = 64
    public static let maxTurnFileChanges = 4_096

    public static let artifactChunkBytes = 256 * 1024
    public static let artifactMaxBytes = 20 * 1024 * 1024

    public static let previewCaptureChunkBytes = 256 * 1024
    public static let previewCaptureMaxBytes = 8 * 1024 * 1024
    public static let previewCaptureMaxPixels = 16 * 1024 * 1024
    public static let previewMaxPerSession = 8
    public static let previewLeaseTtlMaxMs = 10 * 60 * 1000
}
